sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -341,6 +342,28 @@ public static Task<WorkspaceStack> CreateOrSelectStackAsync(LocalProgramArgs arg
             this._readyTask = Task.WhenAll(readyTasks);
         }
 
+        private async Task InitializeProjectSettingsAsync(ProjectSettings projectSettings,
+                                                          CancellationToken cancellationToken)
+        {
+            // If given project settings, we want to write them out to
+            // the working dir. We do not want to override existing
+            // settings with default settings though.
+
+            var existingSettings = await this.GetProjectSettingsAsync(cancellationToken);
+            if (existingSettings == null)
+            {
+                await this.SaveProjectSettingsAsync(projectSettings, cancellationToken);
+            }
+            else
```

Comment:
Is there any reason that in the `else` scenario we don't want to do a more robust comparison? I mean you could write an equality comparer for the `ProjectSettings` class and you could straight up check if the `providedSettings != existingSettings` and only throw if it doesn't.

Reply:
lol ignore me. I need to scroll more before commenting. I see you already wrote a comparer. We could use it here though